app/lib/frontend/backend.dart

```diff
@@ -78,14 +78,22 @@ class Backend {
   }
 
   /// Retrieves the names of all packages, ordered by name.
-  Stream<String> allPackageNames({DateTime updatedSince}) {
+  Stream<String> allPackageNames(
+      {DateTime updatedSince, bool excludeDiscontinued: false}) {
     final query = db.query(models.Package);
 
     if (updatedSince != null) {
       query.filter('updated >', updatedSince);
     }
 
-    return query.run().map((p) => (p as models.Package).name);
+    bool isExcluded(models.Package p) =>
```

Comment:
rename this `shouldInclude`, flip the bool – then you can pass it as a lambda into `where` below.

Reply:
I think flipping the bool makes `!excludeDiscontinued || p.isDiscontinued != true` to be harder to read, and passing it as a lambda is not worth changing it.